GameServer/Logic/Maps/SummonersRift.cs

```diff
@@ -325,35 +327,117 @@ public override bool Spawn()
                 MinionSpawnPosition.SPAWN_RED_BOT,
                 MinionSpawnPosition.SPAWN_RED_MID,
             };
-
-            if (_waveNumber < 3)
+            if (_game.GetMap().GetSpawnState())
             {
-                for (var i = 0; i < positions.Count; ++i)
+                if (_waveNumber < 3) //First 3 melee minions spawn whatever situation is
                 {
-                    Minion m = new Minion(_game, _game.GetNewNetID(), MinionSpawnType.MINION_TYPE_MELEE, positions[i], _laneWaypoints[i]);
-                    AddObject(m);
+                    for (var i = 0; i < positions.Count; ++i)
```

Comment:
This should be extracted to it's own function, since now this is copied everywhere that spawns a minion.

Something like
```CS
SpawnMinion(MinionSpawnType type)
{
    var positions = new List<MinionSpawnPosition> { ... }
    for(var i = 0; i < positions.Count; i++)
    {
        var minion = new Minion(_game, _game.GetNewNetID(), type, positions[i], _laneWaypoints[i]);
        AddObject(minion);
    }
}
```